tests/unit/test_issues.py

```diff
@@ -6,109 +6,145 @@
-        self.assertIn('title', actual.keys())
-        self.assertIn('body', actual.keys())
-
-    def test_moderation_template_rejected(self):
-        """Check the return values are for the rejected case."""
-        actual = moderation_template('rejected')
-        self.assertEqual(actual['title'], 'Issue rejected.')
-        self.assertIn('Its original content has been deleted', actual['body'])
-
-    def test_moderation_template_ongoing(self):
-        """Check the return values are for the needsmoderation case."""
-        # test the default
-        actual = moderation_template()
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
-        # test the keyword
-        actual = moderation_template('ongoing')
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
-        # bad keyword, we go back to the default.
-        actual = moderation_template('punkcat')
-        self.assertEqual(actual['title'], 'In the moderation queue.')
-        self.assertIn('put in the moderation queue.', actual['body'])
+FORM = MultiDict([
+    ('browser', 'Firefox 99.0'),
+    ('description', 'sup'),
+    ('details', ''),
+    ('os', 'Mac OS X 13'),
+    ('problem_category', 'unknown_bug'),
+    ('submit_type', ''),
+    ('url', 'http://3139.example.com'),
+    ('username', ''), ])
+
+FORM_ANON = FORM.copy()
+FORM_ANON['submit_type'] = 'github-proxy-report'
+FORM_AUTH = FORM.copy()
+FORM_AUTH['submit_type'] = 'github-auth-report'
+
+
+@pytest.fixture
+def client():
+    """Test client for requests."""
+    webcompat.app.config['TESTING'] = True
+    with webcompat.app.test_client() as client:
+        yield
+
+
+def test_report_with_auth_on(client):
+    """Test that we can still work with github-auth-report."""
+    with patch('webcompat.issues.github.post') as json_resp:
+        json_resp.return_value = {'number': 2}
+        json_data = report_issue(FORM_AUTH, False)
+        assert type(json_data) is dict
+        assert json_data.get('number') == 2
+
+
+def test_report_issue_returns_number(client):
+    """Test issue posting report the right json."""
+    with patch('webcompat.issues.proxy_request') as github_data:
+        github_data.return_value = mock_api_response({
+            'status_code': 201,
+            'content': '[{"number":2}]',
+        })
+        with patch('webcompat.issues.report_private_issue') as silent:
+            # We avoid making a call to report_private_issue
+            silent.return_value = None
+            json_data = report_issue(FORM_ANON, True)
+            assert type(json_data) is dict
+            assert json_data.get('number') == 2
+
+
+def test_report_issue_fails_400_for_unknown_type(client):
+    """Test request fails with a 400 because type unknown."""
+    with pytest.raises(BadRequest):
+        rv = report_issue(FORM, True)
```

Comment:
(same comment here for `report_issue`)

Reply:
see https://github.com/webcompat/webcompat.com/pull/3322#discussion_r433003788